util/testutil/shell.go

```diff
@@ -88,41 +88,79 @@ func (r *TestingReporter) Stderr() io.Writer {
 	return TestingL.Writer()
 }
 
+// LogMonitor manages a list of functions that should scan lines coming from stdout and stderr Readers
+type LogMonitor struct {
+	monitorFuncs map[string]func(string)
+}
+
+// NewLogMonitor creates a LogMonitor for a given pair of stdout and stderr Readers
+func NewLogMonitor(r shell.Reporter, stdout, stderr io.Reader) *LogMonitor {
+	m := &LogMonitor{}
+	m.monitorFuncs = make(map[string]func(string))
+	go m.ScanLog(io.TeeReader(stdout, r.Stdout()))
+	go m.ScanLog(io.TeeReader(stderr, r.Stderr()))
+	return m
+}
+
+// Add registers a new log monitor function
+func (m *LogMonitor) Add(name string, monitorFunc func(string)) error {
```

Comment:
Could we have this accept an interface instead eg: 

```go
type LogMonitor interface {
   Monitor() error
}
```
m.monitorFuncs[name] = m.Monitor